tsl/src/data_node.c

```diff
@@ -1617,6 +1617,11 @@ data_node_alter(PG_FUNCTION_ARGS)
 		PG_RETURN_DATUM(
 			HeapTupleGetDatum(create_alter_data_node_tuple(tupdesc, node_name, server->options)));
 
+	/* drop stale chunks on the unavailable data node, if we are going to
```

Comment:
The location of this code could be after we run `AlterForeignServer`

Reply:
So that means that we run drop_stale_chunks when data node is marked as available, I am not sure this is what we intended. It should conflict with locking, etc, right?

Reply:
I guess your point is to move the check before the call to CommandCounterIncrement()  but after the AlterForeignServer() call